src/main/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSet.java

```diff
@@ -157,14 +157,15 @@ public BQForwardOnlyResultSet(
       throw new BQSQLException("Failed to fetch results. Connection is closed.");
     }
     this.bigquery = bigquery;
-
-    if (completedJob != null) {
-      BiEngineStatistics biEngineStatistics =
-          completedJob.getStatistics().getQuery().getBiEngineStatistics();
-      if (biEngineStatistics != null) {
-        biEngineMode = biEngineStatistics.getBiEngineMode();
-        biEngineReasons = biEngineStatistics.getBiEngineReasons();
-      }
+    Optional<BiEngineStatistics> biEngineStatisticsOptional =
```

Comment:
Had to go read how Optional works, think I understand it now... so you're using Optional to chain calls through completedJob via `map`. (So I guess map isn't passing nulls on and you're guaranteed your BiEngineStatistics is present and can extract the remaining properties. Ok, cool.

Reply:
Props to @bobdxiao for the fix!